spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationBuilder.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright 2018-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.retrytopic;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.classify.BinaryExceptionClassifier;
+import org.springframework.classify.BinaryExceptionClassifierBuilder;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.core.KafkaOperations;
+import org.springframework.kafka.retrytopic.destinationtopic.DestinationTopic;
+import org.springframework.kafka.retrytopic.destinationtopic.DestinationTopicPropertiesFactory;
+import org.springframework.kafka.support.AllowDenyCollectionManager;
+import org.springframework.retry.backoff.BackOffPolicy;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.backoff.ExponentialRandomBackOffPolicy;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.backoff.NoBackOffPolicy;
+import org.springframework.retry.backoff.SleepingBackOffPolicy;
+import org.springframework.retry.backoff.UniformRandomBackOffPolicy;
+import org.springframework.util.Assert;
+
+/**
+ *
+ * Builder class to create {@link RetryTopicConfiguration} instances.
+ *
+ * @author Tomaz Fernandes
+ * @since 2.7
+ *
+ */
+public class RetryTopicConfigurationBuilder {
+	private int maxAttempts = BackOffValuesGenerator.NOT_SET;
+	private BackOffPolicy backOffPolicy;
```

Comment:
One more style point, sorry, we add a blank line before each field (including the first).

Reply:
Done